tests/components/geo_location/test_usgs_earthquakes_feed.py

```diff
@@ -0,0 +1,196 @@
+from tests.common import assert_setup_component, async_fire_time_changed
+import homeassistant.util.dt as dt_util
+
+CONFIG = {
+    geo_location.DOMAIN: [
+        {
+            'platform': 'usgs_earthquakes_feed',
+            CONF_FEED_TYPE: 'past_hour_m25_earthquakes',
+            CONF_RADIUS: 200
+        }
+    ]
+}
+
+CONFIG_WITH_CUSTOM_LOCATION = {
+    geo_location.DOMAIN: [
+        {
+            'platform': 'usgs_earthquakes_feed',
+            CONF_FEED_TYPE: 'past_hour_m25_earthquakes',
+            CONF_RADIUS: 200,
+            CONF_LATITUDE: 15.1,
+            CONF_LONGITUDE: 25.2
+        }
+    ]
+}
+
+
+def _generate_mock_feed_entry(external_id, title, distance_to_home,
+                              coordinates, place=None,
+                              attribution=None, time=None, updated=None,
+                              magnitude=None, status=None,
+                              entry_type=None, alert=None):
+    """Construct a mock feed entry for testing purposes."""
+    feed_entry = MagicMock()
+    feed_entry.external_id = external_id
+    feed_entry.title = title
+    feed_entry.distance_to_home = distance_to_home
+    feed_entry.coordinates = coordinates
+    feed_entry.place = place
+    feed_entry.attribution = attribution
+    feed_entry.time = time
+    feed_entry.updated = updated
+    feed_entry.magnitude = magnitude
+    feed_entry.status = status
+    feed_entry.type = entry_type
+    feed_entry.alert = alert
+    return feed_entry
+
+
+async def test_setup(hass):
+    """Test the general setup of the platform."""
+    # Set up some mock feed entries for this test.
+    mock_entry_1 = _generate_mock_feed_entry(
+        '1234', 'Title 1', 15.5, (-31.0, 150.0),
+        place='Location 1', attribution='Attribution 1',
+        time=datetime.datetime(2018, 9, 22, 8, 0,
+                               tzinfo=datetime.timezone.utc),
+        updated=datetime.datetime(2018, 9, 22, 9, 0,
+                                  tzinfo=datetime.timezone.utc),
+        magnitude=5.7, status='Status 1', entry_type='Type 1',
+        alert='Alert 1')
+    mock_entry_2 = _generate_mock_feed_entry(
+        '2345', 'Title 2', 20.5, (-31.1, 150.1))
+    mock_entry_3 = _generate_mock_feed_entry(
+        '3456', 'Title 3', 25.5, (-31.2, 150.2))
+    mock_entry_4 = _generate_mock_feed_entry(
+        '4567', 'Title 4', 12.5, (-31.3, 150.3))
+
+    with patch('geojson_client.usgs_earthquake_hazards_program_feed.'
+               'UsgsEarthquakeHazardsProgramFeed') as mock_feed:
+        mock_feed.return_value.update.return_value = 'OK', [mock_entry_1,
+                                                            mock_entry_2,
+                                                            mock_entry_3]
+
+        utcnow = dt_util.utcnow()
+        # Patching 'utcnow' to gain more control over the timed update.
+        with patch('homeassistant.util.dt.utcnow', return_value=utcnow):
```

Comment:
Multiple patches can be applied in a single context manager.

https://github.com/home-assistant/home-assistant/blob/dev/tests/components/deconz/test_init.py#L23-L24

Reply:
OK